tools/decode.py

```diff
@@ -37,100 +45,62 @@ def run(file, **kwargs):
-                                    # decreasing
-                                    state = 'dec2stop'
-                            elif round(servo.goalPosition) == 0:
-                                # First half
-                                if round(servo.goalPosition) > prev_goal_position:
-                                    # increasing
-                                    state = 'inc1stop'
-                                else:
-                                    # decreasing
-                                    state = 'dec1stop'
-                            elif round(servo.goalPosition) > prev_goal_position:
-                                # increasing
-                                state = 'inc'
-                            else:
-                                # decreasing
-                                state = 'dec'
-
-                            prev_goal_position = round(servo.goalPosition)
-
-                        c_pass.append({
-                            'error_flags': servo.errorFlags,
-                            'present_position': servo.presentPosition,
-                            'goal_position': servo.goalPosition,
-                            'present_speed': servo.presentSpeed,
-                            'goal_speed': servo.movingSpeed,
-                            'load': servo.load,
-                            'voltage': servo.voltage,
-                            'temperature': servo.temperature
-                        })
-
-                    else:
-                        out = re.sub(r'\s+', ' ', MessageToJson(msg, True))
-                        out = '{{ "type": "{}", "timestamp": {}, "data": {} }}'.format(parsers[type_hash][0].decode('utf-8'), timestamp, out)
-                        # Print as a json object
-                        print(out)
 
+        # NBS File Format:
+        # 3 Bytes - NUClear radiation symbol header, useful for synchronisation when attaching to a existing stream.
+        # 4 Bytes - The remaining packet length i.e. 16 bytes + N payload bytes
+        # 8 Bytes - 64bit timestamp in microseconds. Note: this is not necessarily a unix timestamp.
+        # 8 Bytes - 64bit bit hash of the message type.
+        # N bytes - The binary packet payload.
+
+        # While we can read a header
+        while len(f.read(3)) == 3:
+
+            # Read our size
+            size = struct.unpack('<I', f.read(4))[0]
+
+            # Read our payload
+            payload = f.read(size)
+
+            # Read our timestamp
+            timestamp = struct.unpack('<Q', payload[:8])[0]
+
+            # Read our hash
+            type_hash = payload[8:16]
+
+            # If we know how to parse this type, parse it
+            if type_hash in decoders:
+                msg = decoders[type_hash][1].FromString(payload[16:])
+
+                # Outputting an image as json would be bad
+                if decoders[type_hash][0] == b'message.input.Image':
+                    pass
```

Comment:
print without the data? otherwise we won't know they exist in the file. i.e.

```
out = re.sub(r'\s+', ' ', MessageToJson(msg, True))
out = '{{ "type": "{}", "timestamp": {} }}'.format(decoders[type_hash][0], timestamp)
print(out)
```